src/app/manager-dashboard/reports/reports-detail.component.ts

```diff
@@ -37,7 +37,7 @@ export class ReportsDetailComponent {
 
   getLoginActivities() {
     this.activityService.getLoginActivities(this.planetCode).subscribe(visits => {
-      this.reports.visits = visits.slice(0, 5);
+      this.reports.visits = visits.filter( visit => visit.user !== '').slice(0, 5);

```

Comment:
After moving to `reports.service.ts` we should also add the condition `visit.user !== undefined`. Same for line 53.